cmd/kube-controller-manager/app/core.go

```diff
@@ -71,14 +71,27 @@ func startServiceController(ctx ControllerContext) (bool, error) {
 }
 
 func startNodeController(ctx ControllerContext) (bool, error) {
-	_, clusterCIDR, err := net.ParseCIDR(ctx.Options.ClusterCIDR)
-	if err != nil {
-		glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+	clusterCIDR := &net.IPNet{}
+	var err error
+	if len(ctx.Options.ClusterCIDR) != 0 {
+		_, clusterCIDR, err = net.ParseCIDR(ctx.Options.ClusterCIDR)
+		if err != nil {
+			glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+		}
+	} else {
+		clusterCIDR = nil
```

Comment:
Sorry, when ClusterCIDR is "", I just set clusterCIDR to nil, because nodecontroller will check it, its code as following:
if clusterCIDR == nil {
			glog.Fatal("NodeController: Must specify clusterCIDR if allocateNodeCIDRs == true.")
		}

On the other hand, is it better to change the code of nodecontroller?